src/launch/policy.c

```diff
@@ -999,64 +1046,42 @@ int policy_export(Policy *policy, sd_bus_message *m) {
                         return error_origin(r);
         }
 
-        r = sd_bus_message_open_container(m, 'r', "buu(" POLICY_T_BATCH ")");
-        if (r < 0)
-                return error_origin(r);
+        {
+                uid_t prev_uid = 0;
 
-        r = sd_bus_message_append(m, "buu", false, 0, SYSTEMUIDMAX);
-        if (r < 0)
-                return error_origin(r);
+                if (n_at_console_uids > 0) {
 
-        r = sd_bus_message_open_container(m, 'r', POLICY_T_BATCH);
-        if (r < 0)
-                return error_origin(r);
+                        qsort(at_console_uids, n_at_console_uids, sizeof(*at_console_uids), policy_uid_compare);
 
-        r = policy_export_connect(policy, NULL, &policy->no_console_entries.connect_list, m);
-        r = r ?: policy_export_own(policy, NULL, &policy->no_console_entries.own_list, m);
-        r = r ?: policy_export_xmit(policy, NULL, &policy->no_console_entries.send_list, m);
-        r = r ?: policy_export_xmit(policy, NULL, &policy->no_console_entries.recv_list, m);
-        if (r)
-                return error_trace(r);
+                        while (at_console_uids[n_at_console_uids - 1] > SYSTEMUIDMAX)
+                                --n_at_console_uids;
+                }
 
-        r = sd_bus_message_close_container(m);
-        if (r < 0)
-                return error_origin(r);
+                for (size_t i = 0; i < n_at_console_uids; ++i) {
+                        r = policy_export_console(policy, m, &policy->no_console_entries, prev_uid, at_console_uids[i]);
+                        if (r < 0)
+                                return error_trace(r);
 
-        r = sd_bus_message_close_container(m);
-        if (r < 0)
-                return error_origin(r);
+                        r = policy_export_console(policy, m, &policy->at_console_entries, at_console_uids[i], at_console_uids[i] + 1);
+                        if (r < 0)
+                                return error_trace(r);
 
-        r = sd_bus_message_open_container(m, 'r', "buu(" POLICY_T_BATCH ")");
-        if (r < 0)
-                return error_origin(r);
+                        prev_uid = at_console_uids[i];
```

Comment:
We could coalesce consequetive entries here. But... eh... not worth it?

Reply:
My thoughts exactly.